src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -0,0 +1,214 @@
+
+        public override int GetHashCode()
+        {
+            var currentHash = GetHashCodeNoParameters();
+            foreach (var param in _parameters)
+            {
+                currentHash = Hash.Combine(param.MethodHashCode(), currentHash);
+            }
+            return currentHash;
+        }
+
+        internal int GetHashCodeNoParameters()
+            => Hash.Combine(ReturnType, Hash.Combine(CallingConvention.GetHashCode(), RefKind.GetHashCode()));
+
+        internal override CallingConvention CallingConvention { get; }
+        public override bool ReturnsVoid => ReturnTypeWithAnnotations.IsVoidType();
+        public override RefKind RefKind { get; }
+        public override TypeWithAnnotations ReturnTypeWithAnnotations { get; }
+        public override ImmutableArray<ParameterSymbol> Parameters =>
+            _parameters.Cast<FunctionPointerParameterSymbol, ParameterSymbol>();
+        // PROTOTYPE(func-ptr): Implement custom modifiers
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => throw new NotImplementedException();
+        public override MethodKind MethodKind => MethodKind.FunctionPointerSignature;
+
+        internal override DiagnosticInfo? GetUseSiteDiagnostic()
+        {
+            DiagnosticInfo? info = null;
+            CalculateUseSiteDiagnostic(ref info);
+            return info;
+        }
+
+        internal bool GetUnificationUseSiteDiagnosticRecursive(ref DiagnosticInfo? result, Symbol owner, ref HashSet<TypeSymbol> checkedTypes)
+        {
+            return ReturnType.GetUnificationUseSiteDiagnosticRecursive(ref result, owner, ref checkedTypes)
+                || GetUnificationUseSiteDiagnosticRecursive(ref result, RefCustomModifiers, owner, ref checkedTypes)
+                || GetUnificationUseSiteDiagnosticRecursive(ref result, Parameters, owner, ref checkedTypes);
+        }
+
+        public override bool IsVararg => false; // PROTOTYPE(func-ptr): Varargs
+
+        public override Symbol? ContainingSymbol => null;
+        // Function pointers cannot have type parameters
+        public override int Arity => 0;
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+        public override bool IsExtensionMethod => false;
+        public override bool HidesBaseMethodsByName => false;
+        public override bool IsAsync => false;
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+        public override Symbol? AssociatedSymbol => null;
+        public override ImmutableArray<Location> Locations => ImmutableArray<Location>.Empty;
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray<SyntaxReference>.Empty;
+        public override Accessibility DeclaredAccessibility => Accessibility.NotApplicable;
+        public override bool IsStatic => false;
```

Comment:
I can't find an override of RequiresInstanceReceiver, doesn't this mean that other parts of the compiler may expect that calls to this function pointer require an instance receiver?

Reply:
This is just a method signature. The compiler won't be emitting calls to it, it'll be using `calli` with an appropriate calling convention.